src/components/library-camera/screens/capture-photo.js

```diff
@@ -0,0 +1,138 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {Alert, StyleSheet, TouchableOpacity, View} from 'react-native';
+import {RNCamera} from 'react-native-camera';
+import {PERMISSIONS, request, RESULTS} from 'react-native-permissions';
+
+import {container, fs20} from '../../../assets/styles/global.js';
+
+import navigationRoutes from '../../../constants/navigation-routes';
+import {
+  flashOff,
+  flashOn,
+  camera,
+  loop,
+} from '../../../constants/font-codes.js';
+
+import {GoBack, Navigate} from '../../../helpers/routes.js';
+import {IconIcomoon} from '../../shared/custom-icons.js';
+
+const {CREATE_BOOK} = navigationRoutes;
+
+const {
+  FlashMode: {on: flashModeOn, off: flashModeOff},
+  Type: {back: backCamera, front: frontCamera},
+} = RNCamera.Constants;
+
+const {isIOS, emptyArr} = global;
+
+const cameraOptions = {quality: 0.5, base64: true};
+
+const androidCameraPermissionOptions = {
+  title: 'Permission to use camera',
+  message: 'We need your permission to use your camera',
+  buttonPositive: 'Ok',
+  buttonNegative: 'Cancel',
+};
+
+const CapturePhoto = ({navigation}) => {
+  const [cameraAccess, setCameraAccess] = useState(false);
+  const [isFlashOn, setIsFlashOn] = useState(false);
+  const [isCameraTypeBack, setIsCameraTypeBack] = useState(true);
+
+  const cameraRef = useRef(null);
+
+  useEffect(() => {
+    request(isIOS ? PERMISSIONS.IOS.CAMERA : PERMISSIONS.ANDROID.CAMERA).then(
+      result => {
+        if (result == RESULTS.GRANTED) setCameraAccess(true);
+        else {
+          Alert.alert('Requires camera access');
+          GoBack(navigation);
```

Comment:
Shouldnt this run after the click on alert ?

Reply:
Also why are we taking user back ? We should show a persmissions not preset screen